### PR TITLE
Complete tasks script

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44605

## Description
Need a script to wait for and complete tasks

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [x] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        parentPlaybookID=task_parent_id,
+        incidentId=incident_id,
+        input=complete_option
+    )
+    try:
+        raw_response = demisto.executeCommand('taskComplete', args=args)
+        return raw_response[0].get("Contents")
+    except Exception as e:
+        demisto.debug(f'Failed to execute script: taskComplete. Error: {e}')
+        return ''
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:61]: please handle failure of the command gracefully

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    # leave states empty to get all tasks
+    if task_states:
+        args['states'] = ','.join(task_states)
+    try:
+        raw_response = demisto.executeCommand('DemistoGetIncidentTasksByState', args=args)
+        return raw_response[0].get("Contents")
+    except Exception as e:
+        demisto.debug(f'Failed to execute script: DemistoGetIncidentTasksByState. Error: {e}')
+        return []
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:35]: same here

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        if task_name and task_states:
+            raise Exception(f'The task "{task_name}" did not reach the {" or ".join(task_states)} state.')
+        elif task_name:
+            raise Exception(f'The task "{task_name}" was not found by script.')
+        elif task_states:
+            raise Exception(f'None of the tasks reached the {" or ".join(task_states)} state.')
+        else:
+            raise Exception('No tasks were found.')
+
+    return CommandResults(
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:161]: please use tableToMarkdown with `headerTransform=pascalToSpace`

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+commonfields:
+  id: WaitAndCompleteTask
+  version: -1
+enabled: false
+name: WaitAndCompleteTask
+outputs:
+- contextPath: WaitAndCompleteTask.CompletedTask
+  description: 'Task name that was completed by script.'
+  type: String
+- contextPath: WaitAndCompleteTask.FoundTasks
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.yml:75]:
```suggestion
- contextPath: WaitAndCompleteTask.FoundTask
```

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        outputs_prefix='WaitAndCompleteTask',
+        outputs_key_field='',
+        outputs={'CompletedTask': completed_tasks,
+                 'FoundTasks': found_tasks},
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:165]:
```suggestion
                 'FoundTask': found_tasks},
```

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask_test.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+     TASK_BY_STATES, '', [],
+     ['conditional task 1', 'conditional task 2', 'manual task 1', 'manual task 2'])
+])
+def test_wait_and_complete_task_command(mocker, args, tasks_ret_value, complete_res, completed_tasks, found_tasks):
+    """
+    Given:
+        - The "WaitAndCompleteTask" script
+
+    When:
+        - Waiting to all task that has given status
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask_test.py:80]:
```suggestion
        - An incident which runs a playbook with four tasks in a Waiting state.
    When:
        - Calling the "WaitAndCompleteTask" script in the following cases:
            - ...
```
in the "when" and "then" sections please describe the cases.

**Review by dantavori — CHANGES_REQUESTED** [review]
awesome, see my minor notes :)

**Comment by dantavori** [comment]
@idovandijk FYI - please make sure the implementation makes sense for you

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py (hunk 1 above)
Comment (daryakoval) [reply Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:61]: what do you mean ?

Comment (daryakoval) [reply Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:61]: Fixed, hope that I understood you right.

File: Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    args: Dict[str, Any] = {
+        'incidentId': incident_id
+    }
+    # leave states empty to get all tasks
+    if task_states:
+        args['states'] = ','.join(task_states)
+    try:
+        raw_response = demisto.executeCommand('DemistoGetIncidentTasksByState', args=args)
+        return raw_response[0].get("Contents") if raw_response[0].get("Contents") else []
+    except Exception as e:
```
Comment (dantavori) [inline Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py:33]: errors won't raise as exceptions, please use is_error() and get_error()
fix only that (also in complete_task_by_id()), great job 🙏🏽

**Review by dantavori — APPROVED** [review]

**Comment by yaron-libman** [comment]
@julieschwartz18 Please take this over for @tschanfeld - her little guy is in bidud.

**Comment by julieschwartz18** [comment]
@daryakoval - doc review complete. Please review and confirm the comments for the yaml and README. Thanks

**Pull request merged**